### ISSUE TITLE
ealvatag.audio.exceptions.NullBoxIdException

### ISSUE THREAD

**Issue by moldis** [ealva-com/ealvatag#1]
For some m4a files i am getting this exception, when reading

**Comment by pandasys** [comment]
Would you post the full exception text so I can see the stack trace? Also, do you have a small example file that can be checked in under /ealvatag/ealvatag/testdata and used for testing? What is the source of the files that are having this issue? Thanks

**Comment by moldis** [comment]
HI there, thanks for quick response
Here is my file: https://www.dropbox.com/s/rk70rcq7l9uwwzo/19.m4a?dl=0

**Comment by moldis** [comment]
Source of file - file from youtube

**Comment by moldis** [comment]
`ealvatag.audio.exceptions.NullBoxIdException: 

	at ealvatag.audio.mp4.atom.Mp4BoxHeader.<init>(Mp4BoxHeader.java:158)
	at ealvatag.audio.mp4.Mp4MinfBox.<init>(Mp4MinfBox.java:53)
	at ealvatag.audio.mp4.Mp4MdiaBox.<init>(Mp4MdiaBox.java:52)
	at ealvatag.audio.mp4.Mp4TrakBox.<init>(Mp4TrakBox.java:48)
	at ealvatag.audio.mp4.Mp4MoovBox.parse(Mp4MoovBox.java:86)
	at ealvatag.audio.mp4.Mp4MoovBox.<init>(Mp4MoovBox.java:55)
	at ealvatag.audio.mp4.Mp4AudioFileReader.read(Mp4AudioFileReader.java:133)`

**Comment by moldis** [comment]
`Mp4AudioFileReader reader = new Mp4AudioFileReader();
            final AudioFile m4a = reader.read(file, "m4a", false);

            Tag tag = m4a.getTag().or(NullTag.INSTANCE);
            if (tag == NullTag.INSTANCE) {
                tag = m4a.setNewDefaultTag();
            }

            tag.setField(key, value);
            m4a.save();`

This is how i am reading it

**Comment by pandasys** [comment]
I'll take a look when I can. I suspected it might be from YouTube as there have been other reports of problems of this sort. Thanks for the file, that's a big help.

**Comment by moldis** [comment]
Yeah, this m4a is from youtube. Okay thank you for help!

**Comment by pandasys** [comment]
tl;dr - the file tag is not compliant with the file specifications. 

To start investigating this type of issue I try to find another tag reader/media player that can parse the tag. I can't find one which will parse this. Media players can play the audio because they simply skip the malformed tag information and display no meta info, except audio header information (channels, bit rate, sample rate, ...). I tried puddletag and the MusicBrainz Picard app and neither reads the tag info. I haven't looked at the tag of this particular file since the other apps can't read it or write a new tag.

This file type meta information is made up of what they call Boxes. Each box header has an ID that indicates what data can be found in the box. The ID for the incorrect box in this file is all zeros and at that point parsing just bales because the data is malformed and can't be read. I delved into the code at the point where the exception is being thrown and I see no workaround - the file tag appears corrupt or malformed and it's truly an exceptional situation.

 If you can find another tag reader which will parse this info, I can further examine the issue. For now, I'm going to mark this invalid, but if you can find another tag reader which reads this I'll investigate further.

Here's a spec: http://l.web.umkc.edu/lizhu/teaching/2016sp.video-communication/ref/mp4.pdf if you're interested.
